PokemonAPI/PokemonAPI/Middleware/GlobalExceptionMiddleware.cs

```diff
@@ -56,6 +57,42 @@ private async Task HandleExceptionAsync(ILogger logger, HttpContext context, Exc
                     errorDetails.Errors = JsonConvert.SerializeObject(validationEx.ValidationResult);
                 }
 
+                else if (exception is NotFoundException)
+                {
+                    errorDetails.Status = HttpStatusCode.NotFound;
+                    errorDetails.Message = "Not found";
+                }
+
+                else if (exception is CreateException)
+                {
+                    errorDetails.Status = HttpStatusCode.InternalServerError;
+                    errorDetails.Message = "Could not create entity";
+                }
+
+                else if (exception is UpdateException)
+                {
+                    errorDetails.Status = HttpStatusCode.InternalServerError;
+                    errorDetails.Message = "Could not update entity";
+                }
+
+                else if (exception is DeleteException)
+                {
+                    errorDetails.Status = HttpStatusCode.InternalServerError;
+                    errorDetails.Message = "Could not delete entity";
+                }
+
+                else if (exception is UnauthorizedAccessException)
+                {
+                    errorDetails.Status = HttpStatusCode.Unauthorized;
+                    errorDetails.Message = "Wrong username or password";
+                }
```

Comment:
Could you just change this to AuthenticationException instead of UnauthorizedAccessException. My mistake. We are throwing AuthenticationExceptions from our AuthenticationService. Feel free to take a look there.